src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -0,0 +1,173 @@
+               There are other cases where the OutputType will remain the same and still the ability to debuggable a project could change
+               For eg: A project's OutputType could be a Lib and an execution entry point could be added or removed
+               */
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.OutputTypeProperty))
+            {
+                await AddOrRemoveProjectFromStartupProjectList().ConfigureAwait(false);
+            }
+        }
+
+        private async Tasks.Task AddOrRemoveProjectFromStartupProjectList(bool initialize = false)
+        {
+            bool isDebuggable = await IsDebuggable().ConfigureAwait(false);
+
+            if (initialize || isDebuggable != _isDebuggable)
+            {
+                _isDebuggable = isDebuggable;
+                await _threadingService.SwitchToUIThread();
+                if (isDebuggable)
+                {
+                    _startupProjectsListService.AddProject(ref _guid);
+                }
+                else
+                {
+                    _startupProjectsListService.RemoveProject(ref _guid);
+                }
+            }
+        }
+
+        private async Tasks.Task<bool> IsDebuggable()
+        {
+            foreach (var provider in _launchProviders.Value.Debuggers)
+            {
+                // DebugLaunchOptions.StopAtEntryPoint seems like a valid option to use. It works.
+                // Not sure if there is a better option for this scenario.
+                if (await provider.Value.CanLaunchAsync(DebugLaunchOptions.StopAtEntryPoint).ConfigureAwait(true))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _evaluationSubscriptionLink?.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+        #endregion
+
+        // Creating a class which provides the LaunchProviders is a workaround because importing an OrderPrecendenceImportCollection
+        // with 2 Type arguments does not work.
+        [Export]
+        internal class DebuggerLaunchProviders
+        {
+            [ImportingConstructor]
+            public DebuggerLaunchProviders(ConfiguredProject project)
+            {
+                Debuggers = new OrderPrecedenceImportCollection<IDebugLaunchProvider, IDebugLaunchProviderMetadataView>(projectCapabilityCheckProvider: project);
```

Comment:
You should just be able to do this same pattern in the main class, no need for a subclass.